gsr_booking/views.py

```diff
@@ -323,3 +336,121 @@ def book_rooms(self, request, pk):
         )
 
         return Response(result_json)
+
+
+LCW = LibCalWrapper()
+WLW = WhartonLibWrapper()
+
+
+class Locations(generics.ListAPIView):
+    serializer_class = GSRSerializer
+    queryset = GSR.objects.all()
+
+
+class Availability(APIView):
+    """
+    Returns JSON containing all rooms for a given building.
+    Usage:
+        /studyspaces/availability/<building> gives all rooms for the next 24 hours
+        /studyspaces/availability/<building>?start=2018-25-01 gives all rooms in the start date
+        /studyspaces/availability/<building>?start=...&end=... gives all rooms between the two days
+    """
+
+    def get(self, request, lid):
+        start = request.GET.get("start")
+        end = request.GET.get("end")
+        is_wharton = GSR.objects.filter(lid=lid).first().kind == GSR.KIND_WHARTON
+        if is_wharton:
+            response = []
+            gsr = GSR.objects.get(lid=lid)
+            response.append(
+                {
+                    "name": gsr.name,
+                    "gid": gsr.gid,
+                    "rooms": WLW.get_availability(lid, start, end, request.user.username),
+                }
+            )
+            return Response(response)
+        else:
+            response = []
+            rooms = LCW.get_availability(lid, start, end)
+            for category in rooms["categories"]:
+                for room in category["rooms"]:
+                    for availability in room["availability"]:
+                        availability["start_time"] = availability["from"]
+                        availability["end_time"] = availability["to"]
+                        del availability["from"]
+                        del availability["to"]
+            for room in rooms["categories"]:
+                context = {}
+                context["name"] = room["name"]
+                context["gid"] = room["cid"]
+                context["rooms"] = [
+                    {"room_name": x["name"], "id": x["id"], "availability": x["availability"]}
+                    for x in room["rooms"]
+                ]
+                response.append(context)
+            return Response(response)
+
+
+class BookRoom(APIView):
```

Comment:
All of the views from here on should probably have an IsAuthenticated permission check, right?